### PR TITLE
Improve renderSolution function if solution is empty

### PR DESCRIPTION
**Pull request by EMaksy**
# Description

A quick fix to ensure that SaptuneDetails is rendered properly if solutions are empty
If solution is empty then applied and enabled solution will render '-'

## Demo
image
image

## How was this tested? 
Added automated test

### CODE REVIEW

**Review by arbulu89 — COMMENTED** [review]
The solution is good.
Let's add a test and go for it

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx [hunk 1]
```diff
@@ -102,4 +102,36 @@ describe('SaptuneDetails', () => {
...
+      configured_version: configuredVersion,
+      package_version: packageVersion,
+      services,
+      staging,
+      tuning_state: tuningState,
+    } = saptuneStatusFactory.build();
+    const { hostname, id: hostID } = hostFactory.build();
+    renderWithRouter(
+      <SaptuneDetails
+        appliedSolution={solution}
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:119]: I would put `null` directly here. It is more visual

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx [hunk 2]
```diff
@@ -102,4 +102,36 @@ describe('SaptuneDetails', () => {
...
+        enabledSolution={solution}
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:120]: Let's add `enabledNotes` and `appliedNotes` with empty lists, so we check a dash is rendered.
And a "disabled" staging (empty solution and notes)

**Review by arbulu89 — COMMENTED** [review]
Looking good.
I would create a more complete test to check all the "empty" value visualizations.
And update the test name accordingly

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx [hunk 3]
```diff
@@ -110,4 +110,54 @@ describe('SaptuneDetails', () => {
       staging.solutions_ids
     );
   });
+
+  it.each([
+    { solution: null },
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:115]: why all of this strange things?
Except the `null` option the other 2 are not real cases. They won't happen, so we don't need to test them.

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx [hunk 4]
```diff
@@ -110,4 +110,54 @@ describe('SaptuneDetails', () => {
...
+    { solution: { notes: [] } },
+    { solution: undefined },
+  ])(
+    'should display dashes for "Enabled Solution," "Applied Solution," "Staged Notes," and "Staged Solutions" when the solution is null, and staging is disabled',
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:119]: Too long I would say. `should display empty values when no data is coming` would suffice in my opinion

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx [hunk 5]
```diff
@@ -110,4 +110,54 @@ describe('SaptuneDetails', () => {
...
+        tuning_state: tuningState,
+      } = saptuneStatusFactory.build();
+      const { hostname, id: hostID } = hostFactory.build();
+      renderWithRouter(
+        <SaptuneDetails
+          appliedSolution={solution}
+          enabledSolution={solution}
+          hostname={hostname}
+          hostID={hostID}
+          configuredVersion={configuredVersion}
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:138]: Send `enabledNotes` and `appliedNotes` as empty lists and check we display them as "empty"

**Review by arbulu89 — CHANGES_REQUESTED** [review]

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx (hunk 3 above)
Comment (EMaksy) [reply assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:115]: Fair enough, the other cases are irrelevant, i just thought some extra error testing won't harm :D

**Review by arbulu89 — APPROVED** [review]
Noice!
